internal/template/funcmap.go

```diff
@@ -117,7 +118,13 @@ func resourceName(m map[string]interface{}, key string) (string, error) {
 	if !ok {
 		return "", fmt.Errorf("resource name value %v is not a string", v)
 	}
-	return strings.Replace(name, "-", "_", -1), nil
+
+	r := strings.NewReplacer(
```

Comment:
optional you could use the dpt v2 way of doing this too:


```
// invalidIDRE defines the invalid characters not allowed in terraform resource names.
var invalidIDRE = regexp.MustCompile("[^a-z0-9-_]")

func resourceName(id string) string {
	return invalidIDRE.ReplaceAllString(strings.ToLower(id), "_")
}
```

Reply:
Done.